File: system-test/typescript/src/testkit/dsl/core/usecase/commons/system-results.ts
import { success, failure, type Result } from '../../../../common/result.js';
import type { SystemError } from '../../../../driver/port/dtos/errors/SystemError.js';

export class SystemResults {
  private constructor() {
    // Utility class
  }

  static success<T>(value: T): Result<T, SystemError>;
  static success(): Result<void, SystemError>;
  static success<T>(value?: T): Result<T | void, SystemError> {
    return success(value as T);
  }

  static failure<T>(error: SystemError | string): Result<T, SystemError> {
    if (typeof error === 'string') {
      return failure({ message: error, fieldErrors: [] });
    }
    return failure(error);
  }
}

File: system-test/dotnet/Driver.Port/SystemResults.cs
using Common;
using Driver.Port.Dtos.Error;

namespace Driver.Port;

public static class SystemResults
{
    public static Result<T, SystemError> Success<T>(T value)
    {
        return Result<T, SystemError>.Success(value);
    }

    public static Result<VoidValue, SystemError> Success()
    {
        return Result.Success<SystemError>();
    }

    public static Result<VoidValue, SystemError> Failure(string message)
    {
        return Result<VoidValue, SystemError>.Failure(SystemError.Of(message));
    }

    public static Result<VoidValue, SystemError> Failure(SystemError error)
    {
        return Result<VoidValue, SystemError>.Failure(error);
    }

    public static Result<T, SystemError> Failure<T>(string message)
    {
        return Result<T, SystemError>.Failure(SystemError.Of(message));
    }

    public static Result<T, SystemError> Failure<T>(SystemError error)
    {
        return Result<T, SystemError>.Failure(error);
    }
}

